splink/blocking.py

```diff
@@ -17,9 +20,11 @@ def __init__(
         self,
         blocking_rule,
         salting_partitions=1,
+        sqlglot_dialect=None,
```

Comment:
A more general note on inheritance of the dialect... we may want to think about a more effective strategy for ensuring both our BRs and CLs inherit a sql dialect where possible.

Reply:
Agreed